doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,209 @@
+can each solve a system of linear scalar equations
+
+Here is a simple example of numerically solving one equation:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> nsolve(cos(x) - x, x, 1)
+0.739085133215161
+```
+
+{func}`~.nsolve` calls, and can pass parameters to, 
+mpmath's {external:func}`~mpmath.findroot`.
+
+Overdetermined systems of equations are supported.
+
+## Guidance
+
+### Find complex roots of real function
+
+To solve for complex roots of real functions, specify a nonreal (either purely
+imaginary, or complex) initial point:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 + 2, 1) # Real initial point returns no root
+Traceback (most recent call last):
+    ...
+ValueError: Could not find root within given tolerance. (4.18466446988997098217 > 2.16840434497100886801e-19)
+Try another starting point or tweak arguments.
+>>> from sympy import I
+>>> nsolve(x**2 + 2, I) # Imaginary initial point returns a complex root
+1.4142135623731*I
+>>> nsolve(x**2 + 2, 1 + I) # Complex initial point returns a complex root
+1.4142135623731*I
+```
+
+### Ensure the root found is in a given interval
+
+It is not guaranteed that nsolve will find the root closest to the initial
+point. Here, even though the root `-1` is closer to the initial point of `-0.1`,
+nsolve finds the root `1`:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, -0.1)
+1.00000000000000
+```
+
+You can ensure the root found is in a given interval, if such a root exists,
+using `solver='bisect'` by specifying the interval in a tuple. Here, specifying
+the interval `(-10, 0)` ensures that the root `-1` is found:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, (-10, 0), solver='bisect')
+-1.00000000000000
+```
+
+## Solve a system of equations numerically
+
+### Solve multidimensional functions
+
+To solve multidimensional functions, supply a tuple of
+- functions `(f1, f2)`
+- variables to solve for `(x1, x2)`
+- starting values `(-1, 1)`
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1)))
+Matrix([[-1.19287309935246], [1.27844411169911]])
+```
+
+### Use SciPy on a lambda function *what is the advantage of this method?*
+
+You can *description*
+
+```py
+>>> from sympy.abc import x
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> from scipy import optimize
+>>> f = implemented_function('f', lambda x: x**3 - 1)
+>>> lam_f = lambdify(x, f(x))
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.0
+```
+
+## Use the solution result
+
+### Substitute the result into an expression
+
+You can use [`subs`](sympy.core.basic.Basic.subs) to substitute numerical values
```

Comment:
The best way to do this is to do `expr.n(subs={x: xvalue})` -- or use `evalf` instead of `n` -- per evalf note:
```
Notes
=====

When Floats are naively substituted into an expression,
precision errors may adversely affect the result. For example,
adding 1e16 (a Float) to 1 will truncate to 1e16; if 1e16 is
then subtracted, the result will be 0.
That is exactly what happens in the following:

>>> from sympy.abc import x, y, z
>>> values = {x: 1e16, y: 1, z: 1e16}
>>> (x + y - z).subs(values)
0

Using the subs argument for evalf is the accurate way to
evaluate such an expression:

>>> (x + y - z).evalf(subs=values)
1.00000000000000
```
Since you are getting numerical results and are going to use them in some way, this would be a good habit to form early.